src/actions/validate-pull-request.ts

```diff
@@ -0,0 +1,109 @@
+import { debug, info } from '@actions/core';
+import type { GetResponseDataTypeFromEndpointMethod } from '@octokit/types';
+import type { Config } from '../config';
+import type { PRData } from '../index';
+import { octokit } from '../lib/github';
+import { pluralise } from '../lib/pluralise';
+
+interface Props {
+	pullRequest: GetResponseDataTypeFromEndpointMethod<
+		typeof octokit.pulls.get
+	>;
+	config: Config;
+	prData: PRData;
+}
+
+export const validatePullRequest = async ({
+	pullRequest,
+	config,
+	prData,
+}: Props): Promise<void> => {
+	debug('validatePullRequest');
+
+	info('Checking pull request is valid');
+
+	/*************************************/
+
+	if (
+		!pullRequest.user ||
+		pullRequest.user.login !== config.pullRequestAuthor
+	) {
+		info(
+			`Pull request is not authored by ${config.pullRequestAuthor}, ignoring.`,
+		);
+		return;
```

Comment:
Now that the validate and merge logic is split out, these returns in the validate function won't skip the merge. These are for the cases where the PR isn't a version bump PR so I don't think we should fail, just exit without doing anything else.

Reply:
ah yeah very good point – will look at that today

Reply:
should be resolved by https://github.com/guardian/actions-merge-release-changes-to-protected-branch/pull/30/commits/bee3fe23662df04d13dc428abc0f1839de18ae42

Reply:
Nice 👍